bluecode-pay/src/test/java/com/example/bluecodepay/bluecode/service/BluecodeProviderTest.java

```diff
@@ -0,0 +1,78 @@
+package com.example.bluecodepay.bluecode.service;
+
+import backend.template.entity.Status;
+import com.example.bluecodepay.bluecode.client.BluecodeClient;
+import com.example.bluecodepay.bluecode.entity.BluecodePayment;
+import com.example.bluecodepay.bluecode.response.BluecodeResponse;
+import com.example.bluecodepay.bluecode.response.Result;
+import com.example.bluecodepay.bluecode.transformer.BluecodePaymentTransformer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.ResponseEntity;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.only;
+
+//In process
+@ExtendWith(MockitoExtension.class)
+class BluecodeProviderTest {
+
+    private static final String PAYMENT_ID = "paymentId";
+
+    @Mock
+    private BluecodePaymentTransformer bluecodePaymentTransformer;
+
+    @Mock
+    private BluecodeClient bluecodeClient;
+
+    @Mock
+    private  RetryManager retryManager;
+
+    private BluecodeProvider bluecodeProvider;
```

Comment:
In case if you have dependencies which can be replaced by Mock annotation and not other which can't be replaced by this, you can use InjectMocks annotation